### PR TITLE
[TPI-3699] - Merge new magento

### PR DESCRIPTION
**Pull request by yanoandri**

### CODE REVIEW

**Review by IreneGohtami — APPROVED** [review]

**Comment by candrasaputra** [comment]
don't delete the changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # CHANGELOG
 
+## 2.8.0 (2021-03-30)
```
Comment (candrasaputra) [inline CHANGELOG.md:3]: Since this is big change.
I think better to major version instead of minor version.

3.0.0

**Comment by candrasaputra** [comment]
can we remove logDNA implementation?
since we didn't use it anymore.

File: Xendit/M2Invoice/Helper/ApiRequest.php [hunk 2]
```diff
@@ -41,51 +63,57 @@ public function request(
...
+        $apiKey = $isPublicRequest ? $this->xendit->getPublicApiKey() : $this->xendit->getApiKey();
         $auth = $this->cryptoHelper->generateBasicAuth($apiKey);
 
         $headers = [
             'Authorization' => $auth,
             'Content-Type' => 'application/json',
             'x-plugin-name' => 'MAGENTO2',
             'user-agent' => 'Magento 2 Module',
-            'x-plugin-version' => '2.7.0'
+            'x-plugin-version' => '2.4.0'
```
Comment (candrasaputra) [inline Xendit/M2Invoice/Helper/ApiRequest.php:116]: Please update this version too

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+
+Improvements:
+
+- Upgrade to Enterprise Edition (Require Magento 2.3 & PHP 7.3 or above)
```
Comment (IreneGohtami) [inline CHANGELOG.md:7]: Add compatibility for Magento Enterprise (Require Magento 2.3 & PHP 7.3 or above)

**Comment by IreneGohtami** [comment]
Also please update the readme.md to add these new payment methods:
- DANA
- LinkAja
- Indomaret
- Kredivo
- QRIS
- Direct Debit BRI

File: Xendit/M2Invoice/Controller/Checkout/Redirect.php [hunk 4]
```diff
@@ -136,15 +135,13 @@ public function execute()
             }
 
             $message = 'No action on xendit/checkout/redirect';
```
Comment (IreneGohtami) [inline Xendit/M2Invoice/Controller/Checkout/Redirect.php:137]: $message nya bisa dihapus kalau ga kepake.

File: Xendit/M2Invoice/Helper/ApiRequest.php [hunk 5]
```diff
@@ -66,13 +66,13 @@ public function request(
...
-                $this->magentoCurl->addHeader("Content-Type", $headers['Content-Type']);
-            }
-            if (isset($headers['Authorization'])) {
-                $this->magentoCurl->addHeader("Authorization", $headers['Authorization']);
-            }
+            $this->magentoCurl->addHeader("Content-Type", $headers['Content-Type']);
+            $this->magentoCurl->addHeader("Authorization", $headers['Authorization']);
+            $this->magentoCurl->addHeader("x-plugin-name", $headers['x-plugin-name']);
+            $this->magentoCurl->addHeader("user-agent", $headers['user-agent']);
+            $this->magentoCurl->addHeader("x-plugin-version", $headers['x-plugin-version']);
```
Comment (IreneGohtami) [inline Xendit/M2Invoice/Helper/ApiRequest.php:73]: we can loops the $headers array

**Review by candrasaputra — APPROVED** [review]

**Pull request merged**